### PR TITLE
Make event times an API call, default times based on static variables if fails

### PR DESCRIPTION
**Pull request by PatrickKan**

### CODE REVIEW

File: HackIllinois/Misc/Constants.swift [hunk 1]
```diff
@@ -11,21 +11,65 @@
...
+                        let fridayEndTime = timesDict["fridayEnd"],
+                        let satStartTime = timesDict["saturdayStart"],
+                        let satEndTime = timesDict["saturdayEnd"],
+                        let sunStartTime = timesDict["sundayStart"],
+                        let sunEndTime = timesDict["sundayEnd"]
+                    else {
+                        print("Failed to update times")
+                        return
+                    }
+                    HIConstants.EVENT_START_TIME = eventStartTime
```
Comment (redsn0w422) [inline HackIllinois/Misc/Constants.swift]: My recommendation here is to have a model called HIEventTimes or something where the real times are used, and you can keep the constants as hardcoded below for fallback if the times don't come through (and switch to using the HIEventTImes)

File: HackIllinois/Misc/Constants.swift [hunk 2]
```diff
@@ -11,21 +11,65 @@
...
+
+    func updateTimes() {
+
+        // Update the times of event
+        TimeService.getTimes()
+            .onCompletion { result in
+                do {
+                    let (times, _) = try result.get()
+                    let timesDict = times.data
+                    guard let eventStartTime = timesDict["eventStart"],
```
Comment (redsn0w422) [inline HackIllinois/Misc/Constants.swift]: Parsing would be better if you had a Time struct that was decodeable and just return that directly, Swift will parse it into the struct for you.

File: HackIllinois/Misc/Constants.swift (hunk 1 above)
Comment (PatrickKan) [reply HackIllinois/Misc/Constants.swift]: Made HITimeDataSource

File: HackIllinois/Misc/Constants.swift (hunk 2 above)
Comment (PatrickKan) [reply HackIllinois/Misc/Constants.swift]: Done

**Comment by PatrickKan** [comment]
Needed to create the constructor like that in the Time struct since I can't import HackIllinois to get HIConstants in the Time struct due to circular dependency

File: HackIllinois/Misc/Constants.swift [hunk 3]
```diff
@@ -11,6 +11,8 @@
 //
 
 import Foundation
+import CoreData
+import HIAPI
```
Comment (redsn0w422) [inline HackIllinois/Misc/Constants.swift]: Looks like these are extra/unused?

File: HackIllinois/DataSources/HITimeDataSource.swift [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+//
+
+import Foundation
+import HIAPI
+
+final class HITimeDataSource {
+    static var shared = HITimeDataSource()
+
+    public static let defaultTimes = EventTimes(
+        eventStart: HIConstants.EVENT_START_TIME,
```
Comment (redsn0w422) [inline HackIllinois/DataSources/HITimeDataSource.swift]: Can we move the constants into `HITimeDataSource`?

File: HackIllinois/DataSources/HITimeDataSource.swift [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+        // Update the times of event
+        TimeService.getTimes()
+            .onCompletion { result in
+                do {
+                    let (timeContainer, _) = try result.get()
+                    self.eventTimes = timeContainer.eventTimes
+                    success = true
+                } catch {
+                    self.eventTimes = HITimeDataSource.defaultTimes
```
Comment (redsn0w422) [inline HackIllinois/DataSources/HITimeDataSource.swift]: Is this needed?

File: HackIllinois/DataSources/HITimeDataSource.swift [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+                    print(error)
```
Comment (redsn0w422) [inline HackIllinois/DataSources/HITimeDataSource.swift]: Can you use OSLogger here instead?

File: HackIllinois/Misc/Constants.swift (hunk 3 above)
Comment (PatrickKan) [reply HackIllinois/Misc/Constants.swift]: HIAPI is used for the call to get the times from api

File: HackIllinois/ViewControllers/HIHomeViewController.swift [hunk 7]
```diff
@@ -166,6 +166,7 @@ extension HIHomeViewController {
 extension HIHomeViewController: HICountdownViewControllerDelegate {
     func countdownToDateFor(countdownViewController: HICountdownViewController) -> Date? {
         let now = Date()
+
```
Comment (redsn0w422) [inline HackIllinois/ViewControllers/HIHomeViewController.swift]: Accidental new line?

**Pull request merged**